Restructure Docs

Andre601 opened the pull request:
Restructures the current documentation.

Things left to do:

- [ ] Add redirects for old pages to new ones
- [ ] Add Git Integration page content
- [ ] Add GitLab Integration page content
- [ ] Add BitBucket Integration page content
- [ ] Add pages suggested in this PR

gmitch215 commented:
My apologies. I thought I tried to submit a review, not a commit...

gmitch215 commented on docs/faq/adding-your-project.md:
```diff
@@ -0,0 +1,12 @@
+You are only required to make one join request with a project.  
+If you are accepted to the CI, you can create new projects by yourself without additional approval.
+
+With that said, abusing this freedom will have severe punishments as a result.
```
Maybe be specific about what they are?

Andre601 replied:
I'll probs add a mention of the guidelines and its punishments section.

Andre601 replied:
Improved the note.

gmitch215 commented on docs/faq/build-jdk-8-project.md:
```diff
@@ -0,0 +1,31 @@
+# How do I build a Maven Project with JDK 8?
+
+Jenkins dropped official support for JDK 8 Maven builds. To still build projects using this Java version can you either use our custom Maven plugin to select a JDK Toolchain based on your `maven-compiler-plugin` settings or cross-compile with a newer JDK Version.
```
Most of my suggested changes are based on what Grammarly tells me what to do.
```suggestion
Jenkins dropped official support for JDK 8 Maven builds. To still build projects using this Java version, you can either use our custom Maven plugin to select a JDK Toolchain based on your `maven-compiler-plugin` settings, or cross-compile with a newer JDK Version.
```

gmitch215 commented on docs/faq/build-jdk-8-project.md:
```diff
@@ -0,0 +1,31 @@
+
+## Toolchain Plugin
+
+Our custom Maven Plugin can select a toolchain (e.g. JDK 8) during the build process. For this the following workaround is required:
```
```suggestion
Our custom Maven Plugin can select a toolchain (e.g. JDK 8) during the build process. For this, the following workaround is required:
```

gmitch215 commented on docs/faq/non-mc-projects.md:
```diff
@@ -0,0 +1,6 @@
+# Are non-Minecraft projects allowed?
+
+Yes.
```
I would be explicit about what kinds of non-MC projects we accept (Minecraft clones, completely different games, corporate-level apps, etc), or at the very least, be explicit about what kinds of projects we **don't** accept.

Andre601 replied:
I don't feel like this is a necessary thing here. At most could I say it needs to be a Java project that complies with our Usage Guidelines, but either giving examples of what we accept or not accept could leave open some holes for people to later abuse... Tho, that's my view of things.

Andre601 replied:
Should hopefully be clearer now

gmitch215 commented on docs/services/ci/index.md:
```diff
@@ -0,0 +1,59 @@
+# CI Service
+CodeMC allows you to configure steps for your project to execute whenever a new build is triggered.  
+It also has integrated support for automatic builds from various Remote Repository hosts such as GitHub, GitLab or Gitea/Forgejo.
```
Maybe just say any Git provider?
```suggestion
It also has integrated support for automatic builds from all Remote Repository hosts that support [Git](https://git-scm.com).
```

gmitch215 commented on docs/services/ci/integrations/setup-github.md:
```diff
@@ -0,0 +1,42 @@
+CodeMC supports integrating with [GitHub](https://github.com){ target="_blank" rel="nofollow" } for features such as creating new builds on commits or return a build status on Pull requests, commits, etc.  
+This page tries to explain how you can enable such features.
+
+## Setup automatic Builds
```
```suggestion
## Setup Automatic Builds
```

gmitch215 commented on docs/services/ci/setups/gradle-job.md:
```diff
@@ -0,0 +1,59 @@
+/// info | This is optional
+///
+
+We recommend enabling **Add timestamps to the Console Output**
```
```suggestion
We recommend enabling **Add timestamps to the Console Output**.
```

gmitch215 commented on docs/services/ci/setups/gradle-job.md:
```diff
@@ -0,0 +1,59 @@
+
+## Final Steps
+
+1.  Press the **Save** Button to save your changes and be redirected to your Project's Job page.
```
```suggestion
1.  Press the **Save** Button to save your changes, redirecting you to your Project's Job page.
```

Andre601 replied:
Sounds a bit odd to me... Maybe ", which redirects you to your Project's Job page after."?

gmitch215 replied:
I think future tense is better in this situation, so I changed it to `which will redirect` instead of `which redirects`.

gmitch215 commented on docs/services/ci/setups/maven-job.md:
```diff
@@ -0,0 +1,59 @@
+/// info | This is optional
+///
+
+We recommend enabling **Add timestamps to the Console Output**
```
```suggestion
We recommend enabling **Add timestamps to the Console Output**.
```

gmitch215 commented on docs/services/nexus/deploy.md:
```diff
@@ -0,0 +1,157 @@
+///
+
+## Prerequisites
+Before you can deploy your project to the Nexus will you first need to do some preparations to ensure that everything will work as expected.
```
```suggestion
Before you can deploy your project to the Nexus, you will first need to do some preparations to ensure that everything will work as expected.
```

gmitch215 commented on docs/services/nexus/deploy.md:
```diff
@@ -0,0 +1,157 @@
+    //// warning | Important
+    These text boxes are NOT for your actual username and password. They are used to set the variable names that jenkins should use to get the respective values.
+    ////
+5.  Leave the **Credentials** Menu at **jenkins-deploy**
```
This is the new name for user-specific credentials:
```suggestion
5.  Leave the **Credentials** Menu at **nexus-repository**
```

Andre601 replied:
Is this already implemented? Because last I checked it still was jenkins-deploy.

gmitch215 replied:
`jenkins-deploy` is the name for the global credentials. Because we're not going to remove it (at least not immediately), I changed the name for the new auto-generated user credentials.

gmitch215 commented on docs/usage-guidelines.md:
```diff
@@ -0,0 +1,51 @@
+
+### Project must be of use for others
+While it can be nice to have a CI for your library to upload and share, should you limit it to libraries and projects that can have a use for others.  
+Effectively, if the project's purpose is to ease your coding of other Projects, while not having any use outside of that for others, is it not something you can and should put on our CI.
```
```suggestion
Effectively, if the project's purpose is to ease your coding of other Projects, while not having any use outside of that for others, is it something you shouldn't put on our CI.
```

gmitch215 commented on docs/usage-guidelines.md:
```diff
@@ -0,0 +1,51 @@
+### Project should have an english documentation available
+CodeMC accepts non-english projects. However, to allow a user to easily understand what this project does and is about, should it provide easy access to an english documentation of the Project's README or other noteworthy documentation a user may want to read.
+
+Translations done through a translator without proofreading by an english speaking person should have a note of not being 100% accurate and done via translators.
```
```suggestion
Translations done through a translator without proofreading by an English-speaking person should note less than 100% accuracy, indicating the use of a translator.
```

gmitch215 commented on docs/usage-guidelines.md:
```diff
@@ -0,0 +1,51 @@
+- Removal of your account(s).
+- Complete blocking of access to our CI and/or Nexus.
+
+CodeMC may issue one or multiple of the above punishments at any given time at our own discretion.
```
I usually like to keep things explicit and clear whenever punishments are involved.
```suggestion
CodeMC may issue one or multiple of the above punishments at any given time at our discretion. Any of these actions may happen in any order of any severity. 
```

gmitch215 commented on docs/usage-guidelines.md:
```diff
@@ -0,0 +1,51 @@
+CodeMC may issue one or multiple of the above punishments at any given time at our own discretion.
+
+## Guideline Updates and Opting out
+These guidelines may be updated at any given moment. We may notify users on our [Discord Server](https://discord.gg/AGcFMu6) about updates to these Guidelines, do not guarantee this however.  
```
Updating rules without proper notifications sounds lazy. I would explicitly state that notifications are for updates that matter.
```suggestion
These guidelines may be updated at any given moment. We will notify users on our [Discord Server](https://discord.gg/AGcFMu6) about significant updates to these Guidelines.
```

Andre601 replied:
I wouldn't guarantee this, especially since it would make it a difficult task, as we would now need to decide on what is a "significant" change worth notifying people.
I have my doubts we will actually make many updates to these guidelines anyways, so I'll leave this as it is right now.

gmitch215 replied:
Gotcha. I think "significant" could be defined as an impact that changes the way we would usually expect people to interact with the service (e.g. we now require 2FA on GitHub accounts)

gmitch215 commented on mkdocs.yml:
```diff
@@ -1,5 +1,5 @@
-site_name: 'Documentation'
-site_description: 'Wiki about the CI of CodeMC'
+site_name: 'CodeMC Documentation'
+site_description: 'Official Wiki of CodeMC'
```
Maybe a better description?
```suggestion
site_description: 'Official Wiki for the CodeMC Project'
```

gmitch215 reviewed (changes requested):
70% Grammer, 30% Changes

gmitch215 commented on docs/faq/non-mc-projects.md:
```diff
@@ -1,6 +1,6 @@
 
-CodeMC accepts a wide range of Projects for all kinds of topics, games, etc.  
-Only important part is, that the project in question does not violate our [Usage Guidelines](../usage-guidelines.md) to be allowed.
+**Long Answer:** CodeMC accepts any Java project that shows a level of quality and usefulness for the overall Java Community, be it a library, a software or even a game.  
```
I feel excluded as a Kotlin user.

In all seriousness, I would implicitly include projects that are related to Java instead of explicitly saying "Java Community" to include projects that are technically "Java Projects" but don't have Java sources (e.g. Kotlin, Groovy, Scala, Clojure, etc). Also includes minor Grammarly changes.
```suggestion
**Long Answer:** CodeMC accepts any Java (or Java-related) project that shows thorough quality and usefulness for the overall programming community, be it a library, software, or even a game. If you're using Kotlin or Scala, your project should compile to the JVM while continuing to provide services intended for the community.
```

gmitch215 commented:
Didn't edit everything else, only added the missing content from the `TODO` pages.

Information was borrowed from official documentation. BitBucket isn't implemented yet until @sgdc3 installs the BitBucket plugin. Feel free to directly change the files.

sgdc3 commented:
@gmitch215 bitbucket integration installed 👍

Andre601 commented:
I've now set up redirects for the old pages under the `/jenkins/` subpage, to allow them still working in case they have been shared in the past.

One thing that is still left is updating the commit/build status integration for GitHub, as that one uses info that by the looks of it do no longer work with the current Jenkins implementation of CodeMC. I sadly don't know what the current/new aproach would be, if there still is one.

gmitch215 reviewed (approved):
If @sgdc3 has nothing else, LGTM

Andre601 commented on docs/services/ci/integrations/setup-gitea.md:
```diff
@@ -0,0 +1,20 @@
+4. Add `https://ci.codemc.io/...` as the target URL.
+5. Under **Trigger on** either leave it at `push events` or chose `Custom events...` and choose the following:
+    - `Pull request`
+    - `Pushe`
```
Need to fix this typo. Making a suggestion to commit it directly.
```suggestion
    - `Push`
```

Andre601 commented:
Made some final changes on my ends now. Namely I added the git-revision-date-localized and the git-committers plugins to use.
First will add the date of a page's last edit and its creation using the git history, which is useful for pages like the usage guidelines (See image below).
Second adds avatars of the users who edited the pages using GitHub API. Found this to be a nice way to give credists to contributors. It's disabled for environments without the right env vars set (See mkdocs.yml for the ones used).

![grafik](https://github.com/user-attachments/assets/49f77dbc-98f7-4915-80e6-5dde5a61b806)

I also updated the GitHub Actions Workflow to now use a direct deployment to GitHub pages which is usually faster and better to do. It will require to save the custom subdomain again in the repo settings so GitHub is aware of it being used as the CNAME file has now usage here using this aproach.
The `gh-pages` branch can also be removed after this.

gmitch215 commented on docs/usage-guidelines.md:
```diff
@@ -0,0 +1,63 @@
+## Guideline Updates and Opting out
+
+These guidelines may be updated at any given moment and we will inform about those on our [Discord Server](https://discord.gg/AGcFMu6) unless not possible for technical reasons.  
+Should we update these Guidelines will we grant a 1 Wekk (7 Days) Period to Opt out of these Guidelines. You may Opt out by no longer using our Services. By continuing to using our services after the aforementioned time Period are you agreeing to the updated Guidelines.
```
Missed typos.
```suggestion
Should we update these Guidelines will we grant a 1 Week (7 Days) Period to Opt out of these Guidelines. You may Opt-out by no longer using our Services. By continuing to use our services after the aforementioned period are you agreeing to the updated Guidelines.
```

gmitch215 commented on docs/services/nexus/deploy.md:
```diff
@@ -0,0 +1,157 @@
+    //// warning | Important
+    These text boxes are NOT for your actual username and password. They are used to set the variable names that jenkins should use to get the respective values.
+    ////
+5.  Leave the **Credentials** Menu at **jenkins-deploy**
```
```suggestion
5.  Leave the **Credentials** Menu at **nexus-repository**
```
